src/ServiceControl/Monitoring/HeartbeatMonitor.cs

```diff
@@ -0,0 +1,53 @@
+namespace ServiceControl.Monitoring
+{
+    using System;
+    using System.Threading;
+
+    class HeartbeatMonitor
+    {
+        volatile RecordedHeartbeat heartbeat = new RecordedHeartbeat(HeartbeatStatus.Unknown, null);
```

Comment:
R# complains that this usage of `volatile` with the two cases of `ref` below don't play nicely together. I've done some research on this but I'd appreciate more eyes.